crates/swc_css_parser/src/lexer/mod.rs

```diff
@@ -1007,19 +996,19 @@ where
         maybe_third: Option<char>,
     ) -> LexResult<bool> {
         // Look at the first code point:
-        let first = maybe_first.or(self.input.cur());
+        let first = maybe_first.or(self.cur());
 
         match first {
             // U+002B PLUS SIGN (+)
             // U+002D HYPHEN-MINUS (-)
             Some('+') | Some('-') => {
-                match maybe_second.or(self.input.peek()) {
+                match maybe_second.or(self.next()) {
```

Comment:
I think this will call `self.next()` unconditionally.

Reply:
We rewrite logic from looking ahead on current looking as in spec, i.e. we use
- `self.cur()` 
- `self.next()`
- `self.next_next()`

https://www.w3.org/TR/css-syntax-3/#starts-with-a-number

> This section describes how to check if three code points would start a number. The algorithm described here can be called explicitly with three code points, or can be called with the input stream itself. In the latter case, the three code points in question are the current input code point and the next two input code points, in that order.

Same for other places

Reply:
Got it and fixed